recipes/lief/all/conanfile.py

```diff
@@ -42,80 +47,101 @@ class LiefConan(ConanFile):
-        self._cmake.definitions["LIEF_DISABLE_FROZEN"] = not self.options.with_frozen
-        self._cmake.definitions["LIEF_C_API"] = self.options.with_c_api
-        self._cmake.definitions["LIEF_EXAMPLES"] = False
-        self._cmake.definitions["LIEF_TESTS"] = False
-        self._cmake.definitions["LIEF_DOC"] = False
-        self._cmake.definitions["LIEF_LOGGING"] = False
-        self._cmake.definitions["LIEF_PYTHON_API"] = False
-        self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["LIEF_ART"] = self.options.with_art
+        tc.variables["LIEF_DEX"] = self.options.with_dex
+        tc.variables["LIEF_ELF"] = self.options.with_elf
+        tc.variables["LIEF_OAT"] = self.options.with_oat
+        tc.variables["LIEF_PE"] = self.options.with_pe
+        tc.variables["LIEF_VDEX"] = self.options.with_vdex
+        tc.variables["LIEF_MACHO"] = self.options.with_macho
+        tc.variables["LIEF_ENABLE_JSON"] = self.options.with_json
+        tc.variables["LIEF_DISABLE_FROZEN"] = not self.options.with_frozen
+        tc.variables["LIEF_C_API"] = self.options.with_c_api
+        tc.variables["LIEF_EXAMPLES"] = False
+        tc.variables["LIEF_TESTS"] = False
+        tc.variables["LIEF_DOC"] = False
+        tc.variables["LIEF_LOGGING"] = False
+        tc.variables["LIEF_PYTHON_API"] = False
+        tc.generate()
+
+        deps = CMakeDeps(self)
+        deps.generate()
 
     def build(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-        cmake = self._configure_cmake()
+        apply_conandata_patches(self)
+        cmake = CMake(self)
+        cmake.configure()
         cmake.build()
 
     def package(self):
-        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
-        cmake = self._configure_cmake()
+        copy(self, pattern="LICENSE", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        cmake = CMake(self)
         cmake.install()
-        tools.rmdir(os.path.join(self.package_folder, "share"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
 
     def package_info(self):
-        self.cpp_info.names["pkg_config"] = "lief"
-        self.cpp_info.names["cmake_find_package"] = "LIEF"
-        self.cpp_info.names["cmake_find_package_multi"] = "LIEF"
-        self.cpp_info.libs = tools.collect_libs(self)
-        self.cpp_info.defines = ["_GLIBCXX_USE_CXX11_ABI=1"]
+        self.cpp_info.libs = ["LIEF"]
+
+        self.cpp_info.set_property("cmake_file_name", "LIEF")
+        self.cpp_info.set_property("cmake_target_name", "LIEF::LIEF")
+        self.cpp_info.set_property("pkg_config_name", "LIEF")
```

Comment:
used to be lowercase?

Reply:
official package name seems to be LIEF.
https://lief-project.github.io/doc/latest/installation.html#find-package

I added these lines for conan v2.